app/models/invoice.rb

```diff
@@ -1,5 +1,6 @@
 class Invoice < ApplicationRecord
   validates_presence_of :status
+  enum status: { "in progress" => 0, "cancelled" => 1, "completed" => 2}
```

Comment:
nice job making this a hash, I think it'll leave less room for error down the line